### PR TITLE
CORE-14023 Ledger transaction resolve states

### PR DESCRIPTION
**Pull request by lankydan**
Retrieve ledger transactions in one step, including resolving the input and reference states of the transaction and retrieving the transaction's signatures.

Add `UtxoLedgerPersistenceService.findSignedLedgerTransaction` to retrieve the transaction, resolve its states and the transaction's signatures.

`UtxoSignedLedgerTransaction` was created to wrap behaviour of `UtxoLedgerTransaction` and `UtxoSignedTransaction` into a single interface which assists in our internal code where we need both behaviours.

This change also tidied up and renamed a few classes in the ledger persistence code.

### CODE REVIEW

File: libs/messaging/kafka-message-bus-impl/src/main/kotlin/net/corda/messagebus/kafka/producer/builder/KafkaCordaProducerBuilderImpl.kt [hunk 1]
```diff
@@ -57,6 +57,7 @@ class KafkaCordaProducerBuilderImpl @Activate constructor(
                 val producer = createKafkaProducer(kafkaProperties, onSerializationError)
                 val maxAllowedMessageSize = messageBusConfig.getLong(MessagingConfig.MAX_ALLOWED_MSG_SIZE)
                 val producerChunkService = messagingChunkFactory.createChunkSerializerService(maxAllowedMessageSize)
+//                val producerChunkService = messagingChunkFactory.createChunkSerializerService(12000)
```
Comment (vlajos) [inline libs/messaging/kafka-message-bus-impl/src/main/kotlin/net/corda/messagebus/kafka/producer/builder/KafkaCordaProducerBuilderImpl.kt]: Is this needed here?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt [hunk 2]
```diff
@@ -3,14 +3,41 @@ package net.corda.ledger.persistence.utxo
...
 import net.corda.v5.ledger.common.transaction.CordaPackageSummary
 import net.corda.v5.ledger.utxo.ContractState
 import net.corda.v5.ledger.utxo.StateRef
 import net.corda.v5.ledger.utxo.observer.UtxoToken
 
 interface UtxoPersistenceService {
-    fun findTransaction(id: String, transactionStatus: TransactionStatus): Pair<SignedTransactionContainer?, String?>
+
+    /**
+     * Find a verified signed transaction in the persistence context given it's [id].
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt]:
```suggestion
     * Find a signed transaction in the persistence context given it's [id].
```

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt [hunk 3]
```diff
@@ -3,14 +3,41 @@ package net.corda.ledger.persistence.utxo
...
+     * @param transactionStatus filter for this status.
+     *
+     * @return The found signed transaction and its status, null if it could not be found in the persistence context.
+     *
+     * @throws InconsistentLedgerStateException If the ledger tables are inconsistent.
+     */
+    fun findSignedTransaction(id: String, transactionStatus: TransactionStatus): Pair<SignedTransactionContainer?, String?>
+
+    /**
+     * Find a verified signed ledger transaction in the persistence context given it's [id] and return it with the status it is stored with.
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt]:
```suggestion
     * Find a signed ledger transaction in the persistence context given it's [id] and return it with the status it is stored with.
```

File: libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/transaction/LedgerTransactionContainer.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package net.corda.ledger.utxo.data.transaction
+
+import net.corda.ledger.common.data.transaction.WireTransaction
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.crypto.SecureHash
+
+@CordaSerializable
+data class LedgerTransactionContainer(
```
Comment (vlajos) [inline libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/transaction/LedgerTransactionContainer.kt]: Is not this a container for SignedLedgerTransactions?

File: components/ledger/ledger-utxo-flow/src/test/kotlin/net/corda/ledger/utxo/flow/impl/persistence/UtxoLedgerPersistenceServiceImplTest.kt [hunk 5]
```diff
@@ -173,43 +180,54 @@ class UtxoLedgerPersistenceServiceImplTest {
...
-        assertThat(utxoLedgerPersistenceService.find(testId)).isEqualTo(expectedObj)
+        assertThat(utxoLedgerPersistenceService.findSignedTransaction(testId)).isEqualTo(expectedObj)
 
         verify(serializationService).deserialize<UtxoSignedTransactionInternal>(any<ByteArray>(), any())
         assertThat(argumentCaptor.firstValue).isEqualTo(FindTransactionExternalEventFactory::class.java)
     }
 
     @Test
-    fun `findTransactionWithStatus executes successfully`() {
+    fun `findLedgerTransaction executes successfully`() {
```
Comment (vlajos) [inline components/ledger/ledger-utxo-flow/src/test/kotlin/net/corda/ledger/utxo/flow/impl/persistence/UtxoLedgerPersistenceServiceImplTest.kt]:
```suggestion
    fun `findSignedLedgerTransaction executes successfully`() {
```
?

File: components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/utxo/tests/UtxoPersistenceServiceImplTest.kt [hunk 6]
```diff
@@ -749,4 +803,17 @@ class UtxoPersistenceServiceImplTest {
         val random = Random()
         return SecureHashImpl(DigestAlgorithmName.SHA2_256.name, ByteArray(32).also(random::nextBytes))
     }
+
+    private fun utxoTransactionMetadataExample(cpkPackageSeed: String? = null) = TransactionMetadataImpl(mapOf(
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/utxo/tests/UtxoPersistenceServiceImplTest.kt:807]: Why is this needed to be copied here?

Comment (lankydan) [reply components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/utxo/tests/UtxoPersistenceServiceImplTest.kt:807]: Because the current location of that method depends on `flow-service` and was causing OSGI issues when trying to run the integration tests.

**Review by vlajos — DISMISSED** [review]

**Review by LWogan — APPROVED** [review]
lgtm

**Review by vlajos — APPROVED** [review]

**Review by driessamyn — APPROVED** [review]
